mmdet3d/models/roi_heads/bbox_heads/pv_rcnn_bbox_head.py

```diff
@@ -0,0 +1,489 @@
+            if self.dropout_ratio >= 0 and k in apply_dropout_indices:
+                fc_layers.append(nn.Dropout(self.dropout_ratio))
+        fc_layers = nn.Sequential(*fc_layers)
+        return fc_layers
+
+    def forward(self, feats: torch.Tensor) -> Tuple[torch.Tensor]:
+        """Forward pvrcnn bbox head.
+
+        Args:
+            feats (torch.Tensor): Batch point-wise features.
+
+        Returns:
+            tuple[torch.Tensor]: Score of class and bbox predictions.
+        """
+        rcnn_batch_size = feats.shape[0]
+        feats = feats.permute(0, 4, 1, 2,
+                              3).contiguous().view(rcnn_batch_size, -1, 1)
+        # (BxN, C*6*6*6)
+        shared_feats = self.shared_fc_layer(feats)
+        cls_feats = self.cls_layers(shared_feats)
+        reg_feats = self.reg_layers(shared_feats)
+        bbox_score = self.cls_out(cls_feats).transpose(1,
+                                                       2).contiguous().squeeze(
+                                                           dim=1)  # (B, 1)
+        bbox_reg = self.reg_out(reg_feats).transpose(1,
+                                                     2).contiguous().squeeze(
+                                                         dim=1)  # (B, C)
+        return bbox_score, bbox_reg
+
+    def loss(self, cls_score: torch.Tensor, bbox_pred: torch.Tensor,
+             rois: torch.Tensor, labels: torch.Tensor,
+             bbox_targets: torch.Tensor, pos_gt_bboxes: torch.Tensor,
+             reg_mask: torch.Tensor, label_weights: torch.Tensor,
+             bbox_weights: torch.Tensor) -> Dict:
+        """Coumputing losses.
+
+        Args:
+            cls_score (torch.Tensor): Scores of each roi.
+            bbox_pred (torch.Tensor): Predictions of bboxes.
+            rois (torch.Tensor): Roi bboxes.
+            labels (torch.Tensor): Labels of class.
+            bbox_targets (torch.Tensor): Target of positive bboxes.
+            pos_gt_bboxes (torch.Tensor): Ground truths of positive bboxes.
+            reg_mask (torch.Tensor): Mask for positive bboxes.
+            label_weights (torch.Tensor): Weights of class loss.
+            bbox_weights (torch.Tensor): Weights of bbox loss.
+
+        Returns:
+            dict: Computed losses.
+
+                - loss_cls (torch.Tensor): Loss of classes.
+                - loss_bbox (torch.Tensor): Loss of bboxes.
+                - loss_corner (torch.Tensor): Loss of corners.
+        """
+        losses = dict()
+        rcnn_batch_size = cls_score.shape[0]
+
+        # calculate class loss
+        cls_flat = cls_score.view(-1)
+        loss_cls = self.loss_cls(cls_flat, labels, label_weights)
+        losses['loss_cls'] = loss_cls
+
+        # calculate regression loss
+        code_size = self.bbox_coder.code_size
+        pos_inds = (reg_mask > 0)
+        if pos_inds.any() == 0:
+            # fake a part loss
+            losses['loss_bbox'] = 0 * bbox_pred.sum()
```

Comment:
losses['loss_bbox'] = loss_cls.new_tensor(0) * loss_cls.sum() ?

Reply:
Here is because need to ensure the RCNN model param in the grad backward compute graph.